document/js-api/index.bs

```diff
@@ -1488,20 +1489,24 @@ To <dfn>create a suspending function</dfn> from a JavaScript function |func|, wi
         1. If |ret|.\[[Type]] is <emu-const>throw</emu-const>, then:
             1. Let |type|, |payload| and |opaqueData| be the result of [=coerce a JavaScript exception|coercing the JavaScript exception=] |ret|.\[[Value]].
             1. [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
-        1. Otherwise, if [=list/size=] of |ret| is 1 and [$IsPromise$](|ret|.\[[Value]][0]):
-            1. Let |promise| be |ret|.\[[Value]][0].
-            1. Set the entry for |async_context| in |map| to [=paused=].
-            1. Let |awaitResult| be the result of performing [$Completion$]([$Await$](|promise|)).
-            1. Note: We only invoke [$Await$] if the call to |func| has returned a Promise object.
-            1. Note: This will suspend both this algorithm, and the WebAssembly function being invoked by the [=evaluate a Promising function=] algorithm. On return, |ret| will be either a normal completion or a throw completion.
-            1. If the entry for |async_context| in |map| is not [=paused=] then:
-                1. Perform [=throw a JavaScript exception=] with a {{SuspendError}}.
-            1. Otherwise, set the entry to [=active=].
-            1. If |awaitResult|.\[[Type]] is <emu-const>throw</emu-const>, then:
-               1. Let |type|, |payload| and |opaqueData| be the result of [=coerce a JavaScript exception|coercing the JavaScript exception=] |ret|.\[[Value]].
-               1. [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
-            1. Otherwise, return the result of performing [=coerce a JavaScript return=] on |resultTypes| and |awaitResult|.
-        1. Otherwise, return the result of performing [=coerce a JavaScript return=] on |resultTypes| and |ret|.
+        1. Otherwise,
+            1. Let |prRet| be the result of [$PromiseResolve$]({{%Promise%}},|ret|.\[[Value]]).
+            1. Assert: |prRet|.\[[Type]] is <emu-const>throw</emu-const> or <emu-const>normal</emu-const>.
+            1. If |prRet|.\[[Type]] is <emu-const>throw</emu-const>, then:
+                1. Let |type|, |payload| and |opaqueData| be the result of [=coerce a JavaScript exception|coercing the JavaScript exception=] |ret|.\[[Value]].
+                1. [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
+            1. Otherwise,
+                1. Let |promise| be |prRet|.\[[Value]].
+                1. Set the entry for |async_context| in |map| to [=paused=].
+                1. Let |awaitResult| be the result of performing [$Completion$]([$Await$](|promise|)).
+                1. Note: This will suspend both this algorithm, and the WebAssembly function being invoked by the [=evaluate a Promising function=] algorithm. On return, |ret| will be either a normal completion or a throw completion.
+                1. If the entry for |async_context| in |map| is not [=paused=] then:
+                    1. Perform [=throw a JavaScript exception=] with a {{RuntimeError}}.
```

Comment:
This line changed from a `SuspendError` to `RuntimeError`. Was this intentional?

Reply:
Oh no .. that's the result of github claiming a code conflict... I will fix.